panacea/lib/panacea/pml/analysis/clashes.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Panacea.Pml.Analysis.Clashes do
+  @composite [:task, :sequence, :branch, :selection, :iteration, :action, :process]
+
+  def run(ast) do
+    analyse([], ast)
+    |> dedup_names
+    |> Map.to_list
+    |> Enum.filter(fn {_name, xs} -> length(xs) > 1 end)
+    |> Enum.map(fn {name, conflicts} -> %{name: name, conflicts: conflicts} end)
+  end
+
+  defp analyse(result, {type, attrs, children}) when type in @composite do
```

Comment:
Did you consider using a `Map` as the accumulator instead of a list? Could remove the need to dedup after.

Would look something like this.

```elixir
case Keyword.get?(attrs, :name) do
  nil ->
    result
  name ->
    element = %{name: name, line: .., type: type}
    Map.update(result, name, [element], fn(names) -> [element|names] end)
end
```